ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/RecoveryTest.java

```diff
@@ -676,6 +676,28 @@ public void shouldRecoverDrop() {
     shouldRecover(commands);
   }
 
+  @Test
+  public void shouldRecoverDrop() {
+    server1.submitCommands(
+        "CREATE STREAM A (COLUMN STRING) WITH (KAFKA_TOPIC='A', VALUE_FORMAT='JSON');",
+        "CREATE STREAM B AS SELECT * FROM A;",
+        "DROP STREAM B;"
+    );
+    shouldRecover(commands);
+  }
+
+  @Test
+  public void shouldRecoverDropWithRecreates() {
+    server1.submitCommands(
+        "CREATE STREAM A (COLUMN STRING) WITH (KAFKA_TOPIC='A', VALUE_FORMAT='JSON');",
+        "CREATE STREAM B AS SELECT * FROM A;",
+        "DROP STREAM B;",
```

Comment:
can we test also something that depends on `B` before drop? e.g.
```suggestion
        "CREATE STREAM B AS SELECT * FROM A;",
        "CREATE STREAM C AS SELECT * FROM B;",
        "DROP STREAM C;",
        "DROP STREAM B;",
```